modules/sap_hana/variables.tf

```diff
@@ -114,8 +114,14 @@ variable "network_tags" {
   default     = []
 }
 
+variable "public_ip" {
+  description = "Determines whether a public IP address is added to your VM instance."
+  default     = 1
```

Comment:
This should be of type boolean.

Also to make it a non-breaking change should default to `false`.